gef.py

```diff
@@ -11281,7 +11302,10 @@ def target_remote_posthook():
 
     gef.session.remote = GefRemoteSessionManager("", 0)
     if not gef.session.remote.setup():
-        raise EnvironmentError(f"Failed to create a proper environment for {gef.session.remote}")
+        warn(f"Failed to create a proper environment for {gef.session.remote}")
+        return False
```

Comment:
Raising is better here too, especially since we're in a callback function. Failing `setup`  is not something easily recoverable.

Also need to `None` the session.

```suggestion
        gef.session.remote = None
        raise EnvironmentError("Failed to create a remote environment")
```

Reply:
This function is directly called from gdb:
```
pi if target_remote_posthook(): ...
```
iirc raising an exception here creates a full stack trace in the gdb console.
If this fails, there are many warning that are raised and that helps to know what happened, without disturbing the user (we actually want to warn him that everything will not work as intended, but since we try our best to make everything work as good as possible we don't want to print errors and stacktraces in most situations)